### PR TITLE
Rule Detection Alert Schema

### PR DESCRIPTION
**Pull request by brian-mckinney**
Work in progress for rule_detection alert schema.

https://github.com/elastic/endpoint-dev/issues/8882

### CODE REVIEW

**Comment by brian-mckinney** [comment]
with these updates I am still getting validation errors for the `threat` and `responses` array.
```
DEBUG    root:ecs_helper.py:156 Document failed to pass ECS validator, the following terms did not pass schema check:
    Term: Value - Reason
        responses: [{'@timestamp': '2021-08-13T20:08:31.493036011Z', 'action': {'action': 'kill_process', 'field': 'process.entity_id', 'state': 0}, 'message': 'Success', 'process': {'entity_id': 'YWFhYWFhYWEtYWFhYS1hYWFhLWFhYWEtYWFhYWFhYWFhYWFhLTM0MDg2LTEzMjczMzU4OTExLjgxMzk0MDAw', 'name': 'sleep_9K9P1IOTXH', 'pid': 34086}, 'result': 0}] - Term not found in validation schema
        threat: [{'framework': 'EAF Test', 'tactic': {'id': 'EAF0001', 'name': 'Generic tactic', 'reference': 'https://elastic.co'}, 'technique': [{'id': 'XXXXXX', 'name': 'Process Creation', 'reference': 'https://elastic.co'}]}] - Term not found in validation schema
```

The format of the alert can be seen here: https://gist.github.com/brian-mckinney/8f3eb924a2374f231c75f211b9afb456

**Review by ferullo — CHANGES_REQUESTED** [review]
Shouldn't these fields be `Responses` not `responses` and `rule.Ext.severity` not `rule.severity` since they are not Endpoint extensions to ECS?

File: custom_schemas/custom_rule.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  description: >
+   Rule fields are used to capture the specifics of any observer or agent rules that generate alerts or other notable events.
+
+   Examples of data sources that would populate the rule fields include: network admission control platforms, network or
+   host IDS/IPS, network firewalls, web application firewalls, url filters, endpoint detection and response (EDR) systems, etc.
+
+  type: group
+  fields:
+
+    - name: severity
```
Comment (ferullo) [inline custom_schemas/custom_rule.yml:15]: `event.severity` already exists. What is this severity? It feels confusing to have two severities.

**Comment by brian-mckinney** [comment]
This PR is superceded by https://github.com/elastic/endpoint-package/pull/175

**Pull request closed without merging**